samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -105,7 +91,7 @@ public boolean tryBecomeLeader() {
     if (index == 0) {
       isLeader.getAndSet(true);
       LOGGER.info(zLog("Eligible to become the leader!"));
-      zkLeaderElectorListener.onBecomingLeader(); // inform the caller
+      leaderElectorListener.onBecomingLeader(); // inform the caller
```

Comment:
something like: 
 ZkLeaderElector le = new ZkLeaderElector()
ClasssThatNeedsLeaderElector cle = new ClasssThatNeedsLeaderElector(le)

and the callback will be part of this new class.
It seems to me semanticly more correct way of doing it is passing the callback at the time of trying to become a leader.